src/main/java/dev/misei/html2pdf/controller/ApiController.java

```diff
@@ -1,21 +1,51 @@
 package dev.misei.html2pdf.controller;
 
-import jakarta.servlet.http.HttpServletResponse;
+import dev.misei.html2pdf.application.PdfRenderComponent;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
 
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
 
-public interface ApiController {
+import static dev.misei.html2pdf.model.Constants.SELENIARM;
+import static dev.misei.html2pdf.model.RenderType.TYPE_DATA;
+import static dev.misei.html2pdf.model.RenderType.TYPE_URL;
 
-    default void responseWith(byte[] renderedPdf, HttpServletResponse response) throws IOException {
-        response.setContentType("application/pdf");
-        response.setHeader("Content-Disposition", "attachment; filename=output.pdf");
+/**
+ * To be used with Docker. This requires that you build, compile and use docker-compose
+ */
+@Controller
+@RequestMapping("/docker")
+@Slf4j
+@RequiredArgsConstructor
+public class ApiController {
 
-        try (ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
-             OutputStream responseStream = response.getOutputStream()) {
-            byteStream.write(renderedPdf);
-            byteStream.writeTo(responseStream);
-        }
+    public final PdfRenderComponent renderComponent;
+
+    @GetMapping("/healthCheck")
+    public ResponseEntity<String> healthCheck() {
+        return ResponseEntity.ok("Ok");
+    }
+
+    @GetMapping("/generatePdfUrl")
```

Comment:
It is up to you both to decide. What do you think?

Reply:
Just a tip: For this case is simpler to have a GET because the url can just go as a parameter. In case of the POST, we send the full HTML, so the query load is bigger than in this case. 😄